sim/rogue/shadowstep.go

```diff
@@ -0,0 +1,75 @@
+package rogue
+
+import (
+	"time"
+
+	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/stats"
+)
+
+func (rogue *Rogue) registerShadowstepCD() {
+	if !rogue.Talents.Shadowstep {
+		return
+	}
+
+	actionID := core.ActionID{SpellID: 36554}
+	baseCost := 10 - 5*float64(rogue.Talents.FilthyTricks)
+	var affectedSpells []*core.Spell
+
+	rogue.ShadowstepAura = rogue.RegisterAura(core.Aura{
+		Label:    "Shadowstep",
+		ActionID: actionID,
+		Duration: time.Second * 10,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			for _, spell := range rogue.Spellbook {
+				if spell.Flags.Matches(SpellFlagBuilder | SpellFlagFinisher) {
+					affectedSpells = append(affectedSpells, spell)
+				}
+			}
+		},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			// Damage of your next ability is increased by 20% and the threat caused is reduced by 50%.
+			for _, spell := range affectedSpells {
+				spell.DamageMultiplier *= 1.2
+				spell.ThreatMultiplier *= 0.5
+			}
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			for _, spell := range affectedSpells {
+				spell.DamageMultiplier *= 1 / 1.2
+				spell.ThreatMultiplier *= 1 / 0.5
+			}
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			for _, affectedSpell := range affectedSpells {
+				if spell == affectedSpell {
+					aura.Deactivate(sim)
+				}
+			}
+		},
+	})
```

Comment:
Not for now: but I'd consider handling this in the rotation code, on a per ability basis. this allows a more direct handling, and simplifies e.g. ability priorities